rune/libenclave/attestation/challenger.go

```diff
@@ -2,26 +2,26 @@ package attestation // import "github.com/inclavare-containers/rune/libenclave/a
 
 import (
 	"fmt"
+	"github.com/inclavare-containers/rune/libenclave/attestation/registration"
+	_ "github.com/inclavare-containers/rune/libenclave/attestation/sgx/challenger" // for the registration of sgxEpidChallenger
 )
 
 type Challenger interface {
 	Name() string
 	New(map[string]string) error
 	Check([]byte) error
-	Verify([]byte) (*ReportStatus, error)
-	GetReport([]byte, uint64) (*ReportStatus, map[string]string, error)
-	ShowReportStatus(*ReportStatus)
+	/* the return value significance of Verify(), GetReport(), and ShowReportStatus():
+	   - uint32: statusCode
+	   - interface{}: SpecificStatus
+	*/
+	Verify([]byte) (uint32, interface{}, error)
```

Comment:
I meant ReportStatus is kept in this layer, because the application requires it. The information used to form ReportStatus can be collected from lower layers, as you already did in sgxEpidChallenger.Verify().